clang 13.0 warnings

clang version: 13.0.0 (git@github.com:llvm/llvm-project.git llvmorg-13.0.0-0-gd7b669b3a303) (gcc 4.2.1 compatibility)
```
diskcryptor_common_plug.c:95:17: warning: variable length array folded to constant array as an extension [-Wgnu-folding-constant]
                unsigned char uc[len];
                              ^
1 warning generated.
fake_salts.c:96:36: warning: variable 'pPrep' set but not used [-Wunused-but-set-variable]
        char *cp = split_fields1, *Prep, *pPrep;
                                          ^
1 warning generated.
gpg2john.c:416:10: warning: variable 'gecos_remains' set but not used [-Wunused-but-set-variable]
                        char *gecos_remains = gecos;
                              ^
gpg2john.c:2531:8: warning: variable 'gecos_remains' set but not used [-Wunused-but-set-variable]
        char *gecos_remains = gecos;
              ^
2 warnings generated.
```

--- a/src/diskcryptor_common_plug.c
+++ b/src/diskcryptor_common_plug.c
@@ -90,9 +90,10 @@ unsigned int diskcryptor_iteration_count(void *salt)
 
 int diskcryptor_decrypt_data(unsigned char *key, struct custom_salt *cur_salt)
 {
-	const int len = 96; /* up to 2048 */
+/* up to 2048 */
+#define LEN 96
 	union {
-		unsigned char uc[len];
+		unsigned char uc[LEN];
 		struct dc_header header;
 	} out;
 	int algorithm;
@@ -101,7 +102,7 @@ int diskcryptor_decrypt_data(unsigned char *key, struct custom_salt *cur_salt)
 		unsigned char tweak[16] = { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
 
 		// Note: header.reserved is actually all zeroes but isn't for us because our XTS code is wonky?
-		XTS_decrypt_custom_tweak(key, tweak, out.uc, cur_salt->header, len, 256, algorithm);
+		XTS_decrypt_custom_tweak(key, tweak, out.uc, cur_salt->header, LEN, 256, algorithm);
 #if ARCH_LITTLE_ENDIAN==0
 		out.header.alg_1 = JOHNSWAP(out.header.alg_1);
 		out.header.version = (out.header.version >> 8) | (out.header.version << 8);
@@ -113,4 +114,5 @@ int diskcryptor_decrypt_data(unsigned char *key, struct custom_salt *cur_salt)
 	}
 
 	return 0;
+#undef LEN
 }

